### PR TITLE
docs(benchmarks): refresh large-repo rows on M5 + Starlark/OCI parser robustness fixes

### PR DESCRIPTION
**Pull request by mstykow**
## Summary

- Refreshes the 7 remaining M1-recorded **large-repo** benchmark rows on the M5 Pro host (elasticsearch, nixpkgs, dotnet/runtime, rust, gitlab, linux, chromium) and regenerates the headline chart (median 19.3×, geomean 19.2×, 10k+ gap 36.6×).
- `fix(starlark)`: recover from a missing argument comma in Bazel/Buck files via a conservative post-failure repair, so a single upstream typo no longer costs a file's package extraction (surfaced by chromium's vendored `zstd/lib/BUCK`).
- `fix(oci)`: stop emitting scan errors when matched `manifest.json`/`index.json` files turn out not to be OCI/docker images (surfaced by chromium, ~180 spurious errors on Chrome-extension manifests).

## Scope and exclusions

- Included: benchmark row refresh + the two parser robustness fixes the refresh surfaced. Kept together per request to land all session changes in one PR.
- Explicit exclusions: no methodology change to `--processes` (M5 set stays at 4 for internal consistency; see note below). Two benign residual chromium scan-error classes left as-is (safe binary-content skips on test images; a couple of text-vs-binary `AndroidManifest.xml`/APK fixture edge cases) — out of scope here.

## How to verify

- Parser fixes are covered by new unit/golden tests (`parsers::starlark_parse`, `parsers::oci`, Bazel/Buck recovery tests). Beyond CI: scan a `manifest.json` that isn't an OCI image (e.g. any Chrome-extension manifest) with `--package` and confirm zero `scan_errors`; scan a BUCK/BUILD file with a missing comma between args and confirm packages are still extracted, each carrying `extra_data.provenant_parse_recovery`.
- Benchmark numbers were produced with `xtask compare-outputs --profile common --build-mode optimized` per row; chromium's row was re-measured against the cached ScanCode result after the OCI fix (ScanCode side unchanged).

## Intentional differences from Python

- dotnet/runtime and chromium package counts are **lower** than ScanCode's because Provenant collapses internal Bazel/Buck build targets into one component per build directory (documented in `docs/improvements/bazel-buck-build-targets.md`); ScanCode emits per-target rows (e.g. chromium 962 Bazel + 9 Buck vs Provenant 255 + 3). Real-ecosystem coverage and dependency extraction stay at or above parity.
- nixpkgs: ScanCode's raw dependency total is far larger only because it duplicates `composer.lock` requirements ~400×; Provenant emits more *unique* dependencies without the duplication.

## Follow-up work

- Methodology: M1→M5 row deltas are confounded by core count — ScanCode is ~1.5× faster at 9 proc than 4 proc on this host, while Provenant scales down gracefully. The M5 set is internally consistent at `--processes 4`; a future pass could standardize a higher count for machine utilization.
- Residual chromium scan-error edge cases (text `AndroidManifest.xml` parsed as binary AXML; APKs lacking a manifest) are minor and deferred.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/parsers/starlark_parse.rs [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+        let comprehension = "y = [\n    item\n    for item in source\n]\n";
+        assert_eq!(repair_missing_argument_commas(comprehension), comprehension);
+    }
+
+    #[test]
+    fn test_repair_ignores_brackets_inside_strings() {
+        // A bracket character inside a string must not be treated as structure.
+        let content = "name = \"value with ] bracket\"\nother = 1\n";
+        assert_eq!(repair_missing_argument_commas(content), content);
+    }
```
[the remark this answers is not in the document]
Comment (mstykow) [reply src/parsers/starlark_parse.rs:288]: Good catch — addressed in c08a60ee. Added `test_repair_tracks_strings_inside_bracketed_args`: a string containing `]` inside a depth-1 `func(...)` call, where only correct string tracking keeps depth > 0 so the missing comma is still inserted. The original depth-0 test would pass under broken tracking; this one would not.
